fms/models/__init__.py

```diff
@@ -251,8 +251,11 @@ def get_model(
     hsdp = distributed_strategy == "hsdp"
     fsdp = distributed_strategy == "fsdp"
     ddp = distributed_strategy == "ddp"
-    if (hsdp and local_rank != 0) or ((fsdp or ddp) and rank != 0):
-        initial_device = torch.device("meta")
+    if hsdp or fsdp or ddp:
+        if (hsdp and local_rank != 0) or ((fsdp or ddp) and rank != 0):
+            initial_device = torch.device("meta")
+        else:
+            initial_device = torch.device("cpu")
```

Comment:
why cpu? how does the data eventually get moved to the correct device?

Reply:
FSDP takes care of that, and we want do to do CPU in this case because of big models like Mixtral and Llama 70b

Reply:
we can let the user pick if they want the model > gpu ram or model < gpu ram approach though, rn we assume single checkpoint/layer checkpoint -> fsdp means model > gpu mem

Reply:
ah I see that the original device param is passed to fsdp_wrap